File: algorithms/medium/count_number_of_distinct_integers_after_reverse_operations.cpp
class Solution {
public:
  int countDistinctIntegers(vector<int> &nums) {
    unordered_set<int> s(nums.begin(), nums.end());
    for (auto n : nums) {
      int m = 0;
      while (n > 0) {
        m = m * 10 + n % 10;
        n /= 10;
      }
      s.insert(m);
    }
    return s.size();
  }
};

File: algorithms/medium/count_number_of_distinct_integers_after_reverse_operations.rs
use std::collections::HashSet;

fn re(mut n: i32) -> i32 {
    let mut m = 0;
    while n > 0 {
        m = m * 10 + n % 10;
        n /= 10;
    }
    m
}

impl Solution {
    pub fn count_distinct_integers(nums: Vec<i32>) -> i32 {
        let mut set: HashSet<_> = nums.iter().cloned().collect();
        for n in nums {
            set.insert(re(n));
        }
        set.len() as _
    }
}
